apps/studio/src/features/editing-experience/components/ComplexEditorStateDrawer.tsx

```diff
@@ -200,26 +169,30 @@ export default function ComplexEditorStateDrawer(): JSX.Element {
             status: "error",
           })
 
-          setModifiedAssets(newModifiedAssets)
-          return false
-        }
-
-        return true
+          // NOTE: Do not save page if there are errors uploading assets
+          setModifiedAssets(failedUploads)
+          setPreviewPageState(newPageState)
+        },
       })
 
       if (!isUploadingSuccessful) {
-        // NOTE: Do not save page if there are errors uploading assets
-        setPreviewPageState(newPageState)
         return
       }
 
-      deleteAssets({ fileKeys: assetsToDelete })
+      // Delete the original assets for those that have been modified
+      // This is done by deleting the file key stored in the src attribute, as
+      // it would have been replaced by new file keys after uploading
+      const assetsToDelete = modifiedAssets
+        .map(({ src }) => src?.slice(1))
+        .filter((src) => src !== PLACEHOLDER_IMAGE_FILENAME)
```

Comment:
as discussed offline, maybe we should have a hardcoded default image (served from our own public folder) as a fallback instead of using a hosted image haha. But that can come in a later PR. I worry because then this is a coupled requirement for all our image/ image upload components.


something like
```ts
import { useState } from 'react';
import Image from 'next/image';

export const ImageWithFallback = ({ src, fallbackSrc = "./assets/fallbackimg", ...rest }) => {
    const [imgSrc, setImgSrc] = useState(src);

    return (
        <Image
            {...rest}
            src={imgSrc}
            onError={() => {
                setImgSrc(fallbackSrc);
            }}
        />
    );
};
```

Reply:
Yeah the fallback image is quite straightforward to do inside the components, but right now the placeholder image is being injected by our component selector default blocks definition, and omitting the value is not allowed since the prop is required.

Probably when we can make it optional, then we can remove the need for this placeholder.